homeassistant/components/netatmo/climate.py

```diff
@@ -326,241 +388,140 @@ def device_state_attributes(self):
         if self._battery_level is not None:
             attr[ATTR_BATTERY_LEVEL] = self._battery_level
 
+        if self._model == NA_VALVE:
+            attr[ATTR_HEATING_POWER_REQUEST] = self._room_status.get(
+                "heating_power_request", 0
+            )
+
         return attr
 
     def turn_off(self):
         """Turn the entity off."""
-        if self._module_type == NA_VALVE:
-            self._data.homestatus.setroomThermpoint(
-                self._data.home_id,
-                self._room_id,
-                STATE_NETATMO_MANUAL,
-                DEFAULT_MIN_TEMP,
+        if self._model == NA_VALVE:
+            self._home_status.set_room_thermpoint(
+                self._id, STATE_NETATMO_MANUAL, DEFAULT_MIN_TEMP,
             )
         elif self.hvac_mode != HVAC_MODE_OFF:
-            self._data.homestatus.setroomThermpoint(
-                self._data.home_id, self._room_id, STATE_NETATMO_OFF
-            )
-        self.update_without_throttle = True
+            self._home_status.set_room_thermpoint(self._id, STATE_NETATMO_OFF)
         self.schedule_update_ha_state()
 
     def turn_on(self):
         """Turn the entity on."""
-        self._data.homestatus.setroomThermpoint(
-            self._data.home_id, self._room_id, STATE_NETATMO_HOME
-        )
-        self.update_without_throttle = True
+        self._home_status.set_room_thermpoint(self._id, STATE_NETATMO_HOME)
         self.schedule_update_ha_state()
 
     @property
     def available(self) -> bool:
         """If the device hasn't been able to connect, mark as unavailable."""
         return bool(self._connected)
 
-    def update(self):
-        """Get the latest data from NetAtmo API and updates the states."""
+    @callback
+    def async_update_callback(self):
+        """Update the entity's state."""
         try:
-            if self.update_without_throttle:
-                self._data.update(no_throttle=True)
-                self.update_without_throttle = False
-            else:
-                self._data.update()
-        except AttributeError:
-            _LOGGER.error("NetatmoThermostat::update() got exception")
-            return
+            roomstatus = {}
+
+            self._home_status = self.data_handler.data[self._home_status_class]
+            self._room_status = self._home_status.rooms[self._id]
+            self._room_data = self._data.rooms[self._home_id][self._id]
+
+            roomstatus["roomID"] = self._room_status["id"]
+            if self._room_status["reachable"]:
```

Comment:
The indentation makes this really hard to follow.  Maybe make a new function `_build_room_status`?
```suggestion
            if self._room_status["reachable"]:
                  self._build_room_status(room_status)
```